astro/src/content/blog/forgerock-password-stroge.mdx

```diff
@@ -0,0 +1,114 @@
+---
+publish_date: 2024-02-15
+title: "Understanding the Forgerock Password Storage Scheme"
+description: Understand how Forgerock stores user passwords.
+authors: Mark Robustelli
+image: /img/blogs/forgerock-password-storage/forgerock-password-storage.png
+categories: Education
+tags: passwords, migration, hashing, Forgerock, password hash
+excerpt_separator: "{/* more */}"
+---
+import {RemoteCode} from '@fusionauth/astro-components';
+import Aside from '../../components/Aside.astro';
+
+You have been charged with migrating your authentication system from Forgerock to another platform. How do you move the users without knowing the users' passwords or making them reset it in the new system?
+{/* more */}
+
+The short answer is hashing. In this post you will learn what password hashing is, how Forgerock stores that information and how to use it in a migration.
+
+## What Is Password Hashing?
+
+Password hashing can appear to be a scary phrase. It sounds complicated and high-tech. In truth, the nitty-gritty details can be. Don't fret. The good news is that for the purposes of this post you can think of it in its simplest terms. That would be taking a string of characters, applying an algorithm (or function), and coming up with a different set of characters. 
+
+The algorithm chosen to hash the characters determines the complexity of getting back to the original password. Thankfully, many smart people have figured out a way to hash passwords in a way that it is easy to reproduce a hash if you know the source characters, but very difficult to get to the original set of characters from the hashed characters. This is referred to as One Way Hashing. You may have heard of some of these hashing functions such as SHA-256, PBKDF2, and Bcrypt.
+
+There are two more bits about hashing that you will need to know related to this article. One is the idea of salt and the other is the idea of iterations.
+
+<Aside type="note">
+If you understand the concept of password salt and iterations, feel free to skip to the next section.
+</Aside>
+
+### Salt
+
+You can think of salt as random characters sprinkled on the password before it is run through the algorithm. This is done so that two users using the same password will still have different hashes.
+
+![Add salt to password.](/img/blogs/forgerock-password-storage/password-salt.jpg)
+
+Example:
+
+The hashing function for this example will be to use the next letter of the existing letter. The letter '**a**' will become 'b', the letter '**b**' will become 'c', and so on.
+
+User 'john' chooses '**cat**' for his password and user 'sally' also chooses '**cat**' for her password.
```

Comment:
@mooreds Do you main boring or the way it is introduced?  Do you like:

There are two users, John and Sally.  John chooses...and Sally chooses...

or 

User 'saltbae@example.com' chooses..... and 'cptn-picard@example.com chooses.....

Reply:
Per our slack discussion, I think just referring to the user by their name is fine. That is:

```
John chooses '**cat**' for his password. 
```

Rather than 

```
User 'john' chooses '**cat**' for his password. 
```